### ISSUE TITLE
docs(sphere): notebook + recipe for spherical density estimation

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#55]
## Problem / Request

Add a jupytext-format notebook in `docs/notebooks/sphere_directional_density.py` that demonstrates fitting a sphere flow to a benchmark directional density (the 4-mode VMF mixture from Rezende et al. 2020). Include a `mkdocs.yml` nav entry and verify the executed `.ipynb` renders correctly.

## User Story

> As someone evaluating gauss_flows for a directional / geophysical use case, I want to see a runnable notebook end-to-end that fits a sphere flow to a multimodal target, plots before/after on a Mollweide projection, and reports KL/ESS, so that I can judge whether the library is ready for my problem.

## Motivation

This is the integration smoke test for the wave: it exercises both child issues #B (exp-map flows) and #C (Möbius spline) together with the base distributions from #A. It also doubles as user-facing documentation — the closest analogue we have is `docs/notebooks/torus_circular_flow.py` which shipped with PR #45/#50.

## Demo To Implement

```python
# docs/notebooks/sphere_directional_density.py
# jupytext percent format

# %% [markdown]
# # Density estimation on the 2-sphere
#
# Fits a Möbius-spline flow to a 4-mode Von Mises–Fisher mixture on S².
# Reproduces the benchmark from Rezende et al. 2020 with KL ≤ 0.10.

# %%
from __future__ import annotations
import jax.numpy as jnp
import jax.random as jr
import matplotlib.pyplot as plt
from flowjax.distributions import Transformed
from flowjax.train import fit_to_data
from gauss_flows import (
    UniformOnSphere, VonMisesFisher,
    mobius_spline_flow, emp_flow, emsre_flow,
)

# %% [markdown]
# ## Generate a 4-mode VMF mixture on S²

# %%
modes = jnp.array([
    [ 1.0,  0.0, 0.0],
    [-1.0,  0.0, 0.0],
    [ 0.0,  1.0, 0.0],
    [ 0.0, -1.0, 0.0],
])
kappa = 10.0
# Sample 5000 points from the mixture
data_key = jr.key(0)
mode_idx = jr.choice(data_key, 4, (5000,))
data = jnp.stack([
    VonMisesFisher(modes[k], kappa).sample(jr.fold_in(data_key, i))
    for i, k in enumerate(mode_idx)
])

# %% [markdown]
# ## Mollweide plot of the training data

# %%
# Convert to (lat, lon) for Mollweide
lat = jnp.arcsin(data[:, 2])
lon = jnp.arctan2(data[:, 1], data[:, 0])
fig, ax = plt.subplots(subplot_kw={"projection": "mollweide"}, figsize=(8, 4))
ax.scatter(lon, lat, s=4, alpha=0.4)
ax.set_title("Training data: 4-mode VMF mixture on S²")
ax.grid(True)

# %% [markdown]
# ## Fit a Möbius-spline flow

# %%
flow_key = jr.key(1)
base = UniformOnSphere(d=2)
bijection = mobius_spline_flow(flow_key, n_layers=4, n_spline_bins=8, n_mobius_components=12)
dist = Transformed(base, bijection)
trained_dist, losses = fit_to_data(
    flow_key, dist, data,
    learning_rate=2e-4, max_epochs=200, batch_size=256, val_prop=0.1,
)

# %% [markdown]
# ## Loss curve + flow samples vs data
# (... rest of plotting boilerplate, mirrors the torus notebook structure ...)
```

## Acceptance criteria

- [ ] `docs/notebooks/sphere_directional_density.py` exists in jupytext percent format
- [ ] `uv run --group docs jupytext --to notebook --execute docs/notebooks/sphere_directional_density.py -o docs/notebooks/sphere_directional_density.ipynb` succeeds end-to-end with no errors
- [ ] Final reported KL ≤ 0.10 against the 4-mode VMF target (or document the gap honestly if higher)
- [ ] Mollweide projection plots render inline via `plt.show()` — no `savefig`, no committed PNGs
- [ ] `mkdocs.yml` updated with a nav entry under `Examples`
- [ ] `make docs` succeeds (mkdocs build)
- [ ] Notebook structure mirrors `torus_circular_flow.py` (sections: generate data, plot data, fit flow, loss curve, sample comparison)
- [ ] Both `.py` and executed `.ipynb` committed — the `.ipynb` contains the rendered figure outputs
- [ ] No `plt.switch_backend("Agg")` (per project workflow — figures render inline)

## Bonus (nice to have, not required)

- [ ] Side-by-side comparison of EMP, EMSRE, MS on the same target with KL/ESS table
- [ ] Real-world example beyond synthetic VMF: e.g. fit to the [ERA5 wind direction](https://cds.climate.copernicus.eu/) at a single grid point, or to a public satellite-orientation dataset

## Out of scope

- Hyperparameter sweeps — single set of defaults is fine
- Comparison to non-flow baselines (MCMC samplers etc.)

## Estimate

~150 LOC notebook + small mkdocs.yml change. Small PR.

## Dependencies

Blocked by: #53 (exp-map flows), #54 (Möbius spline)
